arch/src/aarch64/uefi.rs

```diff
@@ -41,3 +45,41 @@ where
         .read_exact_from(guest_addr, uefi_image, uefi_size)
         .map_err(|_| Error::ReadUefiImage)
 }
+
+/// The UEFI binary requires a flash device at address 0.
+/// FlashMemory is a partially simulated flash. It uses memory for the storage,
+/// but doesn't implement the flash device spec.
+pub struct FlashMemory {
+    memory: GuestMemoryMmap,
+}
+
+impl FlashMemory {
+    pub fn new(
+        vm: &Arc<dyn hypervisor::Vm>,
+        addr: GuestAddress,
+        size: usize,
+        slot: u32,
+    ) -> Result<Self> {
+        let memory = GuestMemoryMmap::from_ranges(&[(addr, size)]).unwrap();
+
+        for (_, region) in memory.iter().enumerate() {
```

Comment:
You know this is going to be a single region, therefore it might make more sense to use `GuestRegionMmap::new()` directly.

Reply:
Where can I find some example code of creating a `GuestMemoryMmap` from single region that was created by `GuestRegionMmap::new()`? I am not very familiar with `vm_memory` crate.

Reply:
Something like this should do:

```rust
let region = GuestRegionMmap::new(MmapRegion::new(size).map_err(Error::MmapRegion)?, addr)
            .map_err(Error::GuestRegionMmap)?;
```

Reply:
Thanks. Updated.